TypeCobol/Compiler/CupPreprocessor/CompilerDirectiveBuilder.cs

```diff
@@ -241,41 +241,76 @@ public void EnterSequenceNumberField(List<Token> seqNumField)
-                System.Diagnostics.Debug.Assert(integerLiteralToken.TokenType == TokenType.IntegerLiteral);
-                int current = (int)((IntegerLiteralTokenValue)integerLiteralToken.LiteralValue).Number;
 
-                if (isFirst)
+            foreach(Token token in seqNumField)
+            {
+                if (token.TokenType == TokenType.IntegerLiteral)
                 {
-                    previous = current;
-                    isFirst = false;
+                    
+                    int current = (int)((IntegerLiteralTokenValue)token.LiteralValue).Number;
+
+                    //Report error for negative number and ignore them
+                    if (current < 0)
+                    {
+                        Diagnostic error = new Diagnostic(MessageCode.SyntaxErrorInParser, token.Position(), "Invalid value: " + current + ". Excepted a positive integer.");
+                        CompilerDirective.AddDiagnostic(error);
+                    }
+                    else
+                    {
+                        DeleteDirective.SequenceNumberRange range = new DeleteDirective.SequenceNumberRange();
+                        range.From = current;
+                        range.To = current;
+                        deleteDirective.SequenceNumberRangesList.Add(range);
+                    }
+                    
                 }
-                else
+                else if (token.TokenType == TokenType.UserDefinedWord)
                 {
-                    DeleteDirective.SequenceNumberRange range = new DeleteDirective.SequenceNumberRange();
-                    range.From = previous;
-                    if (current < 0)
+                    var numbers = token.Text.Split('-');
+                    
+                    if (numbers.Length != 2)
                     {
-                        range.To = -current;
-                        isFirst = true;
+                        Diagnostic error = new Diagnostic(MessageCode.SyntaxErrorInParser, token.Position(), "Invalid range format");
+                        CompilerDirective.AddDiagnostic(error);
                     }
                     else
                     {
-                        range.To = previous;
-                        previous = current;
+                        DeleteDirective.SequenceNumberRange range = new DeleteDirective.SequenceNumberRange();
+                        bool isOk = true;
+
+                        if (int.TryParse(numbers[0], out var temp))
+                        {
+                            range.From = temp;
+                        }
+                        else
+                        {
+                            Diagnostic error = new Diagnostic(MessageCode.SyntaxErrorInParser, token.Position(), "Invalid value: " + numbers[0] + ". Excepted a positive integer.");
+                            CompilerDirective.AddDiagnostic(error);
+                            isOk = false;
+                        }
```

Comment:
Create a local function for this to factorize

Reply:
Done in e705b29f87cea7244fae092d8bc7488e8de90876